### PR TITLE
branch flow slacks

### PR DESCRIPTION
**Pull request by jd-lara**
This PR adds the capability to have slacks on the line flows for ACBranches. It doesn't apply to HVDC yet.

### CODE REVIEW

File: src/devices_models/device_constructors/branch_constructor.jl [hunk 1]
```diff
@@ -315,49 +373,114 @@ end
...
 function construct_device!(
     container::OptimizationContainer,
     sys::PSY.System,
     ::ArgumentConstructStage,
-    ::DeviceModel{T, StaticBranchBounds},
+    device_model::DeviceModel{T, StaticBranchBounds},
     ::NetworkModel{<:PM.AbstractPowerModel},
-) where {T <: PSY.ACBranch} end
+) where {T <: PSY.ACBranch}
+    if get_use_slacks(device_model)
```
Comment (daniel-thom) [inline src/devices_models/device_constructors/branch_constructor.jl:433]: duplicate?

File: src/devices_models/devices/AC_branches.jl [hunk 2]
```diff
@@ -289,15 +317,30 @@ function add_constraints!(
...
+        if branch_name ∈ radial_branches_names
             continue
         end
-        for t in time_steps
-            constraint[r[1], t] = JuMP.@constraint(
-                get_jump_model(container),
-                var1[r[1], t]^2 + var2[r[1], t]^2 <= r[2]^2
-            )
+        if use_slacks
+            for t in time_steps
```
Comment (daniel-thom) [inline src/devices_models/devices/AC_branches.jl:330]: This feels like it should be two functions.

File: src/devices_models/devices/AC_branches.jl [hunk 3]
```diff
@@ -543,6 +586,14 @@ function add_constraints!(
     U <: AbstractBranchFormulation,
     V <: PM.AbstractActivePowerModel,
 }
+    error("slacks here")
```
Comment (daniel-thom) [inline src/devices_models/devices/AC_branches.jl:589]: debug?

**Review by daniel-thom — CHANGES_REQUESTED** [review]

File: src/devices_models/devices/AC_branches.jl (hunk 3 above)
Comment (jd-lara) [reply src/devices_models/devices/AC_branches.jl:589]: This is a deeper bug, this code isn't being called.

Comment (jd-lara) [reply src/devices_models/devices/AC_branches.jl]: after testing this is stale code the whole function

File: src/devices_models/devices/AC_branches.jl (hunk 2 above)
Comment (jd-lara) [reply src/devices_models/devices/AC_branches.jl:330]: I would have to duplicate a lot more code or write functions that accept a lot of arguments

**Review by daniel-thom — APPROVED** [review]

**Pull request merged**